torch_xla/csrc/ops/roll.cpp

```diff
@@ -0,0 +1,83 @@
+#include "torch_xla/csrc/ops/roll.h"
+
+#include "tensorflow/compiler/xla/client/lib/constants.h"
+#include "tensorflow/compiler/xla/client/xla_builder.h"
+#include "tensorflow/compiler/xla/xla_client/debug_macros.h"
+#include "tensorflow/compiler/xla/xla_client/util.h"
+#include "torch_xla/csrc/data_ops.h"
+#include "torch_xla/csrc/helpers.h"
+#include "torch_xla/csrc/lowering_context.h"
+#include "torch_xla/csrc/ops/scalar.h"
+
+namespace torch_xla {
+namespace ir {
+namespace ops {
+
+Roll::Roll(const Value& input, std::vector<int64_t> shifts,
+           std::vector<int64_t> dims)
+    : Node(torch::lazy::OpKind(at::aten::roll), {input}, input.xla_shape(),
+           /*num_outputs=*/1, torch::lazy::MHash(shifts, dims)),
+      shifts_(std::move(shifts)),
+      dims_(std::move(dims)) {}
+
+torch::lazy::NodePtr Roll::Clone(OpList operands) const {
+  return ir::MakeNode<Roll>(operands.at(0), shifts_, dims_);
+}
+
+XlaOpVector Roll::Lower(LoweringContext* loctx) const {
```

Comment:
let's create a helper function called `BuildLower` and add it to the https://github.com/pytorch/xla/blob/master/torch_xla/csrc/xla_lower_util.h. `Roll::Lower` can just call the helper function.

Reply:
done